jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/Response.java

```diff
@@ -175,7 +169,17 @@ default void onContent(Response response, Content.Chunk chunk, Runnable demander
      */
     interface AsyncContentListener extends ContentSourceListener
     {
-        void onContent(Response response, Content.Chunk chunk, Runnable demander);
+        /**
+         * Callback method invoked when the response content has been received, parsed and there is demand.
+         * The {@code demander} must be run before this method may be invoked again.
+         *
```

Comment:
Please add to the javadocs that upon return from the method, the chunk is released.
It is implicit in the doctrine, but does not hurt to repeat it (see javadocs of `ContentListener.onContent()`).